src/ethstorage.js

```diff
@@ -134,22 +135,45 @@ export class EthStorage {
         return ethers.getBytes(data);
     }
 
-    async putBlobs(number, data) {
-        if (!data) {
-            throw new Error(`EthStorage: Invalid data.`);
+    async putBlobs(keys, data) {
+        if (!keys || !data) {
+            throw new Error(`EthStorage: Invalid parameter.`);
+        }
+        if (keys.length !== data.length) {
+            throw new Error(`EthStorage: The number of keys and data does not match.`);
+        }
+        if (keys.length > BLOB_COUNT_LIMIT) {
+            throw new Error(`EthStorage: The count exceeds the maximum blob limit.`);
+        }
+
+        const blobLength = keys.length;
+        const blobDataSize = BLOB_DATA_SIZE;
+
+        const blobArr = [];
+        const keyArr = [];
+        const idArr = [];
+        const lengthArr = [];
+        for (let i = 0; i < blobLength; i++) {
+            const d = Buffer.from(data[i]);
+            if (d.length < 0 || d.length > blobDataSize) {
+                throw new Error(`EthStorage: the length of data(Buffer) should be > 0 && < ${blobDataSize}.`);
+            }
+            const blob = encodeBlobs(d);
+            blobArr.push(blob[0]);
+            keyArr.push(ethers.keccak256(stringToHex(keys[i])));
```

Comment:
what is the purpose of using the hash of the key string?

Reply:
It maintains consistency with the key used in other interfaces and supports any content as a key.